packages/elvis/src/components/form/radio-panel.scss

```diff
@@ -0,0 +1,222 @@
+@use '../../utilities/breakpoints';
+@use '../../utilities/mixins';
+@use '../../utilities/outline';
+
+@mixin limit-lines($line-clamp) {
+  overflow: hidden;
+  display: -webkit-box;
+  -webkit-line-clamp: $line-clamp;
+  -webkit-box-orient: vertical;
+}
+
+@mixin square($size) {
+  height: $size;
+  width: $size;
+}
+
+.e-radio-panel-container {
+  --min-width: 150px;
+
+  display: grid;
+  gap: 8px;
+  grid-auto-rows: 1fr; //same height rows
+  place-items: stretch;
+  grid-template-columns: repeat(auto-fill, minmax(var(--min-width), 1fr));
+
+  @include breakpoints.gt-mobile() {
+    gap: 16px;
+    --min-width: 200px;
+
+    &:has(.e-radio-panel--detailed) {
+      --min-width: 250px;
+    }
+  }
+}
+
+.e-radio-panel {
+  --border-color: var(--e-color-border-3);
+  --border-width: 1px;
+
+  align-items: center;
+  border-radius: 100px;
+  box-shadow: inset 0 0 0 var(--border-width) var(--border-color);
+  cursor: pointer;
+  display: grid;
+  gap: 8px;
+  grid-template: auto auto / auto;
+  max-width: 350px;
+  min-width: 150px;
+  padding: 16px 24px;
+
+  @include breakpoints.gt-mobile() {
+    gap: 16px;
+    min-width: 200px;
+    padding: 24px 32px;
+  }
+
+  &:hover,
+  &:has(input:checked) {
+    --border-width: 2px;
+  }
+
+  &:has(input:checked) {
+    --border-color: var(--e-color-border-selected-2);
+  }
+
+  &:has(input:focus-visible) {
+    @include outline.e-outline();
+  }
+
+  .e-radio-panel__icon,
```

Comment:
Overlap the icon container and the icon itself. Then, using visibility hide/show the proper icon during hover.